sweep/sweeper.go

```diff
@@ -454,6 +545,73 @@ func (s *UtxoSweeper) collector(blockEpochs <-chan *chainntnfs.BlockEpoch,
 	}
 }
 
+// bucketForFeeReate determines the proper bucket for a fee rate. This is done
+// in order to batch inputs with similar fee rates together.
+func (s *UtxoSweeper) bucketForFeeRate(
+	feeRate lnwallet.SatPerKWeight) (lnwallet.SatPerKWeight, error) {
+
+	step := lnwallet.SatPerKWeight(s.cfg.FeeRateBucketSize)
+	for bucket := s.relayFeeRate + step; bucket <= s.cfg.MaxFeeRate; bucket += step {
+		if feeRate <= bucket {
+			return bucket, nil
+		}
+	}
+
+	// This should never happen as we assume that the fee rate has already
+	// been validated so that it fits within our constraints, but we'll
+	// return an error anyway.
+	return 0, fmt.Errorf("invalid fee rate of %v", feeRate)
+}
+
+// clusterBySweepFeeRate takes the set of pending inputs within the UtxoSweeper
+// and clusters those together with similar fee rates. Each cluster contains a
+// sweep fee rate, which is determined by calculating the average fee rate of
+// all inputs within that cluster.
+func (s *UtxoSweeper) clusterBySweepFeeRate() []inputCluster {
+	bucketInputs := make(map[lnwallet.SatPerKWeight]pendingInputs)
+	inputFeeRates := make(map[wire.OutPoint]lnwallet.SatPerKWeight)
+
+	// First, we'll group together all inputs with similar fee rates. This
+	// is done by determining the fee rate bucket they should belong in.
+	for op, input := range s.pendingInputs {
+		feeRate, err := s.validateFeePreference(input.feePreference)
```

Comment:
Re-suggesting to add `input.GetFeeRate()`. Sweeper is already a large struct and imo it should be made smaller rather than larger if we change things.